New numpy==2.0.0 breaks runners

From [broken workflow](https://github.com/microsoft/hummingbird/actions/runs/9619950191/job/26537154808):

44 instances like this:
```
___________________ ERROR collecting tests/test_backends.py ____________________
tests/test_backends.py:13: in <module>
    from sklearn.ensemble import GradientBoostingClassifier
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/sklearn/__init__.py:82: in <module>
    from .base import clone
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/sklearn/base.py:17: in <module>
    from .utils import _IS_32BIT
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/sklearn/utils/__init__.py:19: in <module>
    from .murmurhash import murmurhash3_32
sklearn/utils/murmurhash.pyx:1: in init sklearn.utils.murmurhash
    ???
E   ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject

```

It seems other are having similar issues https://github.com/numpy/numpy/issues/26710 , https://github.com/pandas-dev/pandas/issues/59023 https://github.com/scikit-learn/scikit-learn/discussions/29290

--- a/.github/workflows/pythonapp.yml
+++ b/.github/workflows/pythonapp.yml
@@ -59,7 +59,7 @@ jobs:
       run: python -m pip install scikit-learn==1.2.1 numpy==1.26.4
     - name: Pin SKLearn<1.5.0 on not Linux with py3.9
       if: ${{ !(matrix.python-version == '3.9' && startsWith(matrix.os, 'ubuntu') == true) }}
-      run: python -m pip install "scikit-learn<1.5.0 numpy==1.26.4"
+      run: python -m pip install "scikit-learn<1.5.0" numpy==1.26.4
     - name: Run basic tests without extra
       run: pytest
     - name: Coverage on basic tests without extra

--- a/.github/workflows/pythonapp.yml
+++ b/.github/workflows/pythonapp.yml
@@ -56,10 +56,10 @@ jobs:
         python -m pip install -e .[tests] -f https://download.pytorch.org/whl/torch_stable.html
     - name: Test with older SKLearn on Linux with py3.9 to check backward compatibility
       if: ${{ matrix.python-version == '3.9' && startsWith(matrix.os, 'ubuntu') == true }}
-      run: python -m pip install scikit-learn==1.2.1
+      run: python -m pip install scikit-learn==1.2.1 numpy==1.26.4
     - name: Pin SKLearn<1.5.0 on not Linux with py3.9
       if: ${{ !(matrix.python-version == '3.9' && startsWith(matrix.os, 'ubuntu') == true) }}
-      run: python -m pip install "scikit-learn<1.5.0"
+      run: python -m pip install scikit-learn<1.5.0 numpy==1.26.4
     - name: Run basic tests without extra
       run: pytest
     - name: Coverage on basic tests without extra

--- a/.github/workflows/pythonapp.yml
+++ b/.github/workflows/pythonapp.yml
@@ -59,7 +59,7 @@ jobs:
       run: python -m pip install scikit-learn==1.2.1 numpy==1.26.4
     - name: Pin SKLearn<1.5.0 on not Linux with py3.9
       if: ${{ !(matrix.python-version == '3.9' && startsWith(matrix.os, 'ubuntu') == true) }}
-      run: python -m pip install scikit-learn<1.5.0 numpy==1.26.4
+      run: python -m pip install "scikit-learn<1.5.0 numpy==1.26.4"
     - name: Run basic tests without extra
       run: pytest
     - name: Coverage on basic tests without extra

